inference-engine/src/mkldnn_plugin/ngraph_transformations/reshape_prelu.cpp

```diff
@@ -12,27 +12,50 @@
 NGRAPH_RTTI_DEFINITION(MKLDNNPlugin::ReshapePRelu, "ReshapePRelu", 0);
 
 MKLDNNPlugin::ReshapePRelu::ReshapePRelu() {
-    auto prelu = ngraph::pattern::wrap_type<ngraph::opset1::PRelu>({ngraph::pattern::any_input(ngraph::pattern::has_static_shape()),
-                                                                    ngraph::pattern::any_input(ngraph::pattern::has_static_shape())});
+    auto input = ngraph::pattern::any_input(ngraph::pattern::has_static_rank());
+    auto slope = ngraph::pattern::any_input(ngraph::pattern::has_static_rank());
+    auto prelu = ngraph::pattern::wrap_type<ngraph::opset1::PRelu>({ input, slope });
 
     ngraph::matcher_pass_callback callback = [this](ngraph::pattern::Matcher& m) {
         auto prelu = std::dynamic_pointer_cast<ngraph::opset1::PRelu>(m.get_match_root());
-        if (!prelu || ngraph::shape_size(prelu->get_input_shape(1)) == 1 || prelu->get_input_shape(1).size() != 1) {
+
+        const auto prelu_pshape = prelu->input_value(0).get_partial_shape();
+        const auto prelu_rank = prelu_pshape.rank();
+        const auto slope_pshape = prelu->input_value(1).get_partial_shape();
+        const auto slope_rank = slope_pshape.rank();
+        if (!prelu || prelu_rank.is_dynamic() || slope_rank.is_dynamic() || prelu_rank.get_length() == 1 || slope_rank.get_length() != 1) {
```

Comment:
lets simplify these checks as we already have them in the pattern

Reply:
Fixed, thanks!